integration_tests/steps/load_tests.py

```diff
@@ -79,16 +79,19 @@ def check_user_pv_dict_profile(context):
 
 @then('LoadHoursStrategy does not buy energy with rates that are higher than the provided profile')
 def check_user_rate_profile_dict(context):
-    house = next(filter(lambda x: x.name == "House 1", context.simulation.area.children))
-
-    unmatched, unmatched_redis = \
-        ExportUnmatchedLoads(context.simulation.area).get_current_market_results(
-            all_past_markets=True)
-    number_of_loads = 2
+    from integration_tests.steps.integration_tests import get_simulation_raw_results
+    get_simulation_raw_results(context)
+    count = 0
+    unmatched = ExportUnmatchedLoads(context.area_tree_summary_data)
+    for time_slot, core_stats in context.raw_sim_data.items():
+        unmatched_data, _ = unmatched.get_current_market_results(
+            context.area_tree_summary_data, core_stats, time_slot
+        )
+        count += get_number_of_unmatched_loads(unmatched_data)
     # There are two loads with the same final_buying_rate profile that should report unmatched
     # energy demand for the first 6 hours of the day:
-    assert get_number_of_unmatched_loads(unmatched) == int(number_of_loads * 6. * 60 /
-                                                           house.config.slot_length.minutes)
+    number_of_loads = 2
+    assert count == number_of_loads * 6 * 60 / 30
```

Comment:
I assume that you cannot use the GlobalConfig object because d3a is run via the console, right ?

Reply:
Yes that was the reason but since now Im using the Simulation class to launch the simulation, reverting these two lines now